src/scripts/Update.ts

```diff
@@ -916,6 +916,98 @@ class Update implements Saveable {
 
             // Update total proteins obtained to be equal to the total purchased (or whichever is higher)
             saveData.statistics.totalProteinsObtained = Math.max(saveData.statistics.totalProteinsPurchased, saveData.statistics.totalProteinsObtained);
+
+            // Filter already earned milestones due to item/Pokémon name updates
+            const milestones = [
+                [5, '25 x Poké Ball'],
+                [10, '100 x Poké Ball'],
+                [20, '100 x Great Ball'],
+                [30, '100 x Ultra Ball'],
+                [35, '100 x X Click'],
+                [40, '100 x X Attack'],
+                [50, '100 x Small Restore'],
+                [100, 'Deoxys'],
+                [110, '10 x Water Stone'],
+                [120, '10 x Leaf Stone'],
+                [130, '10 x Thunder Stone'],
+                [140, '10 x Fire Stone'],
+                [150, '200 x Medium Restore'],
+                [151, 'Deoxys (Attack)'],
+                [160, '100 x Lucky Egg'],
+                [170, '100 x Lucky Incense'],
+                [180, '100 x Dowsing Machine'],
+                [190, '10 x Mystery Egg'],
+                [200, '100 x Large Restore'],
+                [210, '40 x Water Stone'],
+                [220, '40 x Leaf Stone'],
+                [230, '40 x Thunder Stone'],
+                [240, '40 x Moon Stone'],
+                [250, '6,400 x Ultra Ball'],
+                [251, 'Deoxys (Defense)'],
+                [300, '100 x Linking Cord'],
+                [386, 'Deoxys (Speed)'],
+            ];
+            const highestStageCompleted = saveData.statistics?.battleFrontierHighestStageCompleted || 0;
+            saveData.battleFrontier = {
+                milestones: milestones.filter(([stage]) => stage <= highestStageCompleted),
+                checkpoint: saveData.battleFrontier.checkpoint,
+            };
+
+            // Update Pokemon name changes
+            const renamePokemon = Update.renamePokemonInSaveData;
+            renamePokemon(saveData, 'Bulbasaur (clone)', 'Bulbasaur (Clone)');
```

Comment:
All this is not needed, as the name of the pokemon are no longer in the savefile

Reply:
AFAIK, it's still used in the hatchery/queue

Reply:
Oh, you are totally right... Sorry!